### PR TITLE
check transform job status before deferring SageMakerTransformOperator

### PR DESCRIPTION
**Pull request by Lee-W**
While running a sagemaker transform job in deferrable mode, the condition might already be met before we defer the task into the trigger. This PR intends to check the job status before deferring the task to trigger.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by Lee-W** [comment]
The static check failure should be fixed in https://github.com/apache/airflow/pull/36677

File: airflow/providers/amazon/aws/operators/sagemaker.py [hunk 1]
```diff
@@ -700,8 +700,24 @@ def execute(self, context: Context) -> dict:
             raise AirflowException(f"Sagemaker transform Job creation failed: {response}")
 
         if self.deferrable and self.wait_for_completion:
+            response = self.hook.describe_transform_job(transform_config["TransformJobName"])
+            status = response["TransformJobStatus"]
+            if status in self.hook.failed_states:
+                raise AirflowException(f"SageMaker job failed because {response['FailureReason']}")
+
+            if status == "Completed":
+                self.log.info("%s completed successfully." % self.task_id)
```
Comment (dirrao) [inline airflow/providers/amazon/aws/operators/sagemaker.py]:
```suggestion
                self.log.info("%s completed successfully.", self.task_id)
```

Comment (dirrao) [reply airflow/providers/amazon/aws/operators/sagemaker.py]: The suggested logger format avoids the static check issue.

**Review by dirrao — CHANGES_REQUESTED** [review]

File: airflow/providers/amazon/aws/operators/sagemaker.py (hunk 1 above)
Comment (Lee-W) [reply airflow/providers/amazon/aws/operators/sagemaker.py]: Thanks! I misunderstood how it works. It works now.

**Review by phanikumv — APPROVED** [review]

**Pull request merged**